uaclient/log.py

```diff
@@ -58,3 +63,33 @@ def format(self, record: logging.LogRecord) -> str:
 
         local_log_record["extra"] = extra_message_dict
         return json.dumps(list(local_log_record.values()))
+
+
+def get_user_log_file() -> str:
+    """Gets the correct user log_file storage location
+
+    returns XDG_CACHE_HOME env variable, if it is set,
+    If not set, we return the default path
+    """
+    user = getpass.getuser()
+    user_cache = os.environ.get("XDG_CACHE_HOME")
+    log_file = "{}-{}.log".format(defaults.USER_LOG_FILE_PATH, user)
+    if user_cache:
+        return user_cache + "/" + log_file
+    return pathlib.Path.home().as_posix() + "/.cache/" + log_file
+
+
+def get_all_user_log_files() -> List[str]:
```

Comment:
Right now this will return files that don't exist, and the 10 log limit is in this function.

This function should probably return _all_ user log files that exist.  Then the 10-file limit can go in the `collect_logs` function.

Reply:
Oh makes, sense I thought the the check [here](https://github.com/canonical/ubuntu-pro-client/blob/7da5a991c6143f0650d9e927f3668e56f6df810b/uaclient/actions.py#L215) was enough but we may run having chosen user log files that are not present over present ones. Thanks